File: assignments/scala/space-age/src/test/scala/space-age_test.scala
import org.scalatest._

class SpaceAgeSpecs extends FunSuite with Matchers {
  test ("age in seconds") {
    val age = SpaceAge(1000000)
    age.seconds should be (1000000)
  }

  test ("age in earth years") {
    pending
    val age = SpaceAge(1000000000)
    age.onEarth should be (31.69)
  }

  test ("age in mercury years") {
    pending
    val age = SpaceAge(2134835688)
    age.onEarth should be (67.65)
    age.onMercury should be (280.88)
  }

  test ("age in venus years") {
    pending
    val age = SpaceAge(189839836)
    age.onEarth should be (6.02)
    age.onVenus should be (9.78)
  }

  test ("age on mars") {
    pending
    val age = SpaceAge(2329871239L)
    age.onEarth should be (73.83)
    age.onMars should be (39.25)
  }

  test ("age on jupiter") {
    pending
    val age = SpaceAge(901876382)
    age.onEarth should be (28.58)
    age.onJupiter should be (2.41)
  }

  test ("age on saturn") {
    pending
    val age = SpaceAge(3000000000L)
    age.onEarth should be (95.06)
    age.onSaturn should be (3.23)
  }

  test ("age on uranus") {
    pending
    val age = SpaceAge(3210123456L)
    age.onEarth should be (101.72)
    age.onUranus should be (1.21)
  }

  test ("age on neptune") {
    pending
    val age = SpaceAge(8210123456L)
    age.onEarth should be (260.16)
    age.onNeptune should be (1.58)
  }
}

File: assignments/elixir/space-age/space_age_test.exs
if System.get_env("EXERCISM_TEST_EXAMPLES") do
  Code.load_file("example.exs")
else
  Code.load_file("space_age.exs")
end

ExUnit.start

# You need to define a SpaceAge module containing a function age_on that given a
# planet (:earth, :saturn, etc) and a number of seconds returns the age in years
# on that planet as a floating point number.

defmodule SpageAgeTest do
  use ExUnit.Case, async: true
  doctest SpaceAge

  test "age in earth years" do
    input =  1_000_000_000
    assert_in_delta 31.69, SpaceAge.age_on(:earth, input), 0.005
  end

  test "age_in_mercury_years" do
    input =  2_134_835_688
    assert_in_delta 67.65, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 280.88, SpaceAge.age_on(:mercury, input), 0.005
  end

  test "age_in_venus_years" do
    input = 189_839_836
    assert_in_delta 6.02, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 9.78, SpaceAge.age_on(:venus, input), 0.005
  end

  test "SpaceAge.mars" do
    input = 2_329_871_239
    assert_in_delta 73.83, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 39.25, SpaceAge.age_on(:mars, input), 0.005
  end

  test "SpaceAge.jupiter" do
    input = 901_876_382
    assert_in_delta 28.58, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 2.41, SpaceAge.age_on(:jupiter, input), 0.005
  end

  test "SpaceAge.saturn" do
    input = 3_000_000_000
    assert_in_delta 95.06, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 3.23, SpaceAge.age_on(:saturn, input), 0.005
  end

  test "SpaceAge.uranus" do
    input = 3_210_123_456
    assert_in_delta 101.72, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 1.21, SpaceAge.age_on(:uranus, input), 0.005
  end

  test "SpaceAge.neptune" do
    input = 8_210_123_456
    assert_in_delta 260.16, SpaceAge.age_on(:earth, input), 0.005
    assert_in_delta 1.58, SpaceAge.age_on(:neptune, input), 0.005
  end
end
